medusa/providers/generic_provider.py

```diff
@@ -228,31 +228,40 @@ def find_search_results(self, show, episodes, search_mode, forced_search=False,
         if len(results) == len(episodes):
             return results
 
-        if items_list:
-            # Remove duplicate items
-            items_list_without_dups = []
-            for item in items_list:
-                if item['link'] not in [_['link'] for _ in items_list_without_dups]:
-                    items_list_without_dups.append(item)
-
-            items_list = items_list_without_dups
-
-            # categorize the items into lists by quality
-            items = defaultdict(list)
-            for item in items_list:
-                items[self.get_quality(item, anime=show.is_anime)].append(item)
-
-            # temporarily remove the list of items with unknown quality
-            unknown_items = items.pop(Quality.UNKNOWN, [])
-
-            # make a generator to sort the remaining items by descending quality
-            items_list = (items[quality] for quality in sorted(items, reverse=True))
+        # Remove duplicate items using 'link' as primary key
+        unique_items = {
```

Comment:
@labrys does this mess with the order, as it being converted to a dict?

Reply:
I wasn't overly concerned about order since we are sorting by quality later.  However if you would prefer it to stay in original order that can easily modified to:
```python
unique_items = OrderedDict(
    (item['link'], item)
    for item in items_list
).values()
```

Reply:
Yes please, then I can use it to fix the issue with daily search that I caused.